db: TestCompactionTombstones is flaky

https://github.com/cockroachdb/pebble/actions/runs/3292923536/jobs/5428855283

```console
2022/10/20 21:05:38 error when picking base compaction: file 000000 chosen as seed file for compaction should not be compacting
2022/10/20 21:05:38 error when picking base compaction: file 000000 chosen as seed file for compaction should not be compacting
--- FAIL: TestCompactionTombstones (0.10s)
    compaction_test.go:2135: 
        testdata/compaction_tombstones:315: 000004
        expected:
        num-entries: 3
        num-deletions: 1
        num-range-key-sets: 0
        point-deletions-bytes-estimate: 0
        range-deletions-bytes-estimate: 39
        
        found:
        (not found)
```

Found while working on #2022.

Hit this over here too: https://github.com/cockroachdb/pebble/actions/runs/3293054600/jobs/5429305986

```console
CGO_ENABLED=0 make test TAGS=
2022/10/20 21:28:21 error when picking base compaction: file 000000 chosen as seed file for compaction should not be compacting
2022/10/20 21:28:21 error when picking base compaction: file 000000 chosen as seed file for compaction should not be compacting
--- FAIL: TestCompactionTombstones (0.10s)
    compaction_test.go:2135: 
        testdata/compaction_tombstones:315: 000004
        expected:
        num-entries: 3
        num-deletions: 1
        num-range-key-sets: 0
        point-deletions-bytes-estimate: 0
        range-deletions-bytes-estimate: 39
        
        found:
        (not found)
```

--- a/testdata/compaction_tombstones
+++ b/testdata/compaction_tombstones
@@ -301,7 +301,7 @@ maybe-compact
 # should be deleted. In #90149, the table still had a reference count, and
 # therefore could not be deleted from the filesystem.
 
-define
+define auto-compactions=off
 L6
   rangekey:a-b:{(#1,RANGEKEYDEL)}
   a.SET.2:a

